src/IsoCodes/ZipCode.php

```diff
@@ -21,111 +217,84 @@ public static function validate($zipcode, $country)
         if (empty($zipcode)) {
             return false;
         }
+
+        $country = strtoupper($country);
+        if (isset(self::$patterns[$country])) {
+            $regexp = '/^'.self::$patterns[$country].'$/';
+
+            return (boolean) preg_match($regexp, $zipcode);
+        }
+
         $methodName = 'validate'.trim(ucfirst(strtolower($country)));
         if (!is_callable(array(__CLASS__, $methodName))) {
             throw new \InvalidArgumentException("ERROR: The zipcode validator for $country does not exists yet: feel free to add it.");
         }
-
         return call_user_func_array(array(__CLASS__, $methodName), array($zipcode));
     }
 
     /**
-     * US "ZIP+4" zipcode validator.
-     *
-     * @param string $zipcode
+     * @param $zipcode
      *
      * @return bool
+     * @deprecated use validate($zipcode, 'US')
```

Comment:
Okay but this will also trigger on `validate($zipcode, 'Spain')` so I tweaked the messages to include the correct code. Also fixed the tests.

And some frameworks turn those errors into Exceptions, so not sure how cool that is for everyone, but I changed it.